File: RepoTransBench/target_projects/Python/C++/jtsang4_claude-to-chatgpt/public_tests/test_public_logger.cpp
#include <gtest/gtest.h>
#include <string>
#include "claude_to_chatgpt/logger.h"

TEST(PublicLoggerTest, PublicGetLoggerLevel) {
    Logger logger = get_logger("publicLoggerTest");
    EXPECT_TRUE(logger.is_valid());
    logger.set_level(Logger::Level::ERROR);
    EXPECT_EQ(logger.get_level(), Logger::Level::ERROR);
}

TEST(PublicLoggerTest, PublicGetLoggerName) {
    std::string logger_name = "unique_logger_name";
    Logger logger = get_logger(logger_name);
    EXPECT_EQ(logger.get_name(), logger_name);
}

File: RepoTransBench/target_projects/Python/Rust/jtsang4_claude-to-chatgpt/public_tests/test_public_logger.rs
use claude_to_chatgpt::logger::{get_logger, DummyLogger};
use log::LevelFilter;

#[test]
fn test_public_get_logger_level() {
    let logger = get_logger("publicLoggerTest");
    logger.set_level(LevelFilter::Error);
    assert_eq!(logger.get_level(), LevelFilter::Error);
}

#[test]
fn test_public_get_logger_name() {
    let logger_name = "unique_logger_name";
    let logger = get_logger(logger_name);
    assert_eq!(logger.name(), logger_name);
}